### PR TITLE
Validate functions

### PR DESCRIPTION
**Pull request by shaharyarn**

### CODE REVIEW

File: marine/marine.py [hunk 1]
```diff
@@ -86,6 +86,12 @@ def filter_and_parse(
         self._marine.marine_free(marine_result)
         return success, result
 
+    def validate_bpf(self, bpf: str) -> bool:
+        bpf = bpf.encode("utf-8")
+        if self._marine.validate_bpf(bpf) < 0:
```
Comment (yehonatanz) [inline marine/marine.py]: `<` or `!=`? What's more likely to remain correct?
(Not rehtorical, actually asking)

Comment (yehonatanz) [inline marine/marine.py]: Instead of:
```python
if cond:
    return False
return True
```
use:
```python
return not cond
```

**Review by yehonatanz — CHANGES_REQUESTED** [review]

File: marine/marine.py (hunk 1 above)
Comment (tomlegkov) [reply marine/marine.py]: Why doesn't it return a boolean? I see no applications for error codes here.

**Review by tomlegkov — CHANGES_REQUESTED** [review]

**Review by yehonatanz — APPROVED** [review]

**Review by tomlegkov — APPROVED** [review]
Don't merge until the C part is done :)

**Pull request merged**